operator/cmd/mesh/profile-common.go

```diff
@@ -48,11 +62,24 @@ import (
 // In step 3, the remaining fields in the same user overlay are applied on the resulting profile base.
 // The force flag causes validation errors not to abort but only emit log/console warnings.
 func GenerateConfig(inFilenames []string, setOverlayYAML string, force bool, kubeConfig *rest.Config,
-	l clog.Logger) (string, *v1alpha1.IstioOperatorSpec, error) {
+	l clog.Logger, useClusterIfAvailable bool) (string, *v1alpha1.IstioOperatorSpec, error) {
 
-	fy, profile, err := readYamlProfle(inFilenames, setOverlayYAML, force, l)
-	if err != nil {
-		return "", nil, err
+	var err error
+	var fy, profile string
+	if useClusterIfAvailable && len(inFilenames) == 0 {
+		fy, err = overlayedIOPFromCluster("istio-system", setOverlayYAML, kubeConfig)
```

Comment:
not so sure about hard-coding to `istio-system` here. I think users should be able to pass one in. and if they don't, maybe we list across all namespaces and pick up the namespace from the IOP resource we find?

Reply:
I agree!  The current `istioctl manifest` commands don't take the `--istioNamespace <x>` flag the other commands take.  I propose to create an Issue for that but let this go through with hard-coded namespace.  @ostromart what do you think?

Reply:
sure, sounds ok.

Reply:
if we do this we are going to have yet another namespace flag for install. A few months back we had, if I remember correctly, 8 different options for this. We have tuned it down a bit since then luckily, but would love to not have another one. Can we read the `values.istioNamespace` value instead maybe?

Reply:
My plan is to either make `istioNamespace` in _istioctl/cmd_ public by capitalizing it or passing it in when creating the Cobra command.  We cannot use the `values.istioNamespace` version at this point because we haven't yet read the IOP from the cluster.  We should not take @dgn 's advice and read all of the IOPs, as the user might be doing something weird like having two Istio System namespaces.

If we must make this a parameter let me do it as a follow on, and do it for all of the `manifest` subcommands.